### PR TITLE
Recurse for subqueries in projection plan nodes

### PR DESCRIPTION
**Pull request by wolffcm**
I've backed out my recent fix for ENG-8145 here, and added code to ProjectionPlanNode to recurse into expression trees for both `resolveColumnIndices` (fixes ENG-8173) and `generateOutputSchema` (fixes ENG-8145).  This is consistent with similar code `AbstractScanPlanNode`.

### CODE REVIEW

File: tests/frontend/org/voltdb/regressionsuites/TestSubQueriesSuite.java [hunk 1]
```diff
@@ -1255,6 +1255,41 @@ public void testScalarSubqueryWithOrderByOrGroupBy() throws Exception {
                 expected);
     }
 
+    public void testScalarStarSubqueryWithOrderBy() throws Exception {
```
Comment (jxchinaren) [inline tests/frontend/org/voltdb/regressionsuites/TestSubQueriesSuite.java]: @wolffcm I refactored this test during my merge of fixing FLOAT/VARCHAR expression subquery. That's why you have conflicts here.

Comment (jxchinaren) [inline tests/frontend/org/voltdb/regressionsuites/TestSubQueriesSuite.java]: Looks good to me, except for the tests conflicts.

**Comment by jxchinaren** [comment]
@wolffcm Could you verify ticket ENG-8174 is also fixed? And probably close it also if true.

**Comment by wolffcm** [comment]
I resolved the merge conflicts, and added the test cases from ENG-8174.  I closed ENG-8174 as a dupe of this one.

File: tests/frontend/org/voltdb/regressionsuites/TestSubQueriesSuite.java [hunk 2]
```diff
@@ -1302,6 +1302,80 @@ public void testWhereRowSubSelects() throws NoConnectionsException, IOException,
...
+
+        vt = client.callProcedure("@AdHoc", "select NUM V, (select SUM(RATIO) from R4) from R4 order by V;")
+                .getResults()[0];
+        assertTrue(vt.advanceRow());
+        vt.getLong(0); assertTrue(vt.wasNull());
+        assertEquals(1.1, vt.getDouble(1));
+
+        assertTrue(vt.advanceRow());
+        assertEquals(-1, vt.getLong(0));
+        assertEquals(1.1, vt.getDouble(1));
```
Comment (jxchinaren) [inline tests/frontend/org/voltdb/regressionsuites/TestSubQueriesSuite.java]: Minor: There is only one non-null input row for RATIO, that's probably why this equal assert for double does not need the delta (e.g. 0.0001) difference.

**Comment by jxchinaren** [comment]
Looks good to me as long as the tests are passed. Thanks Chris.

File: tests/frontend/org/voltdb/regressionsuites/TestSubQueriesSuite.java (hunk 2 above)
Comment (wolffcm) [inline tests/frontend/org/voltdb/regressionsuites/TestSubQueriesSuite.java]: Good point.  It's passing so I'll leave it for now.

**Pull request merged**